### ISSUE TITLE
Somehow I cannot upgrade my project's intl dependency to 0.18.0 because of some weird collision with syncfusion_flutter_charts

### ISSUE THREAD

**Issue by MrCsabaToth** [syncfusion/flutter-widgets#1021]
I depend on the latest `syncfusion_flutter_charts: ^20.4.41`. I see that in the syncfusion_flutter_charts's yaml https://github.com/syncfusion/flutter-widgets/blob/master/packages/syncfusion_flutter_charts/pubspec.yaml it depends on `intl: ^0.17.0`. However if I try to bump `intl: ^0.18.0` the `pub get` fails with a message:

> Because syncfusion_flutter_charts 20.4.41 depends on intl ^0.17.0 and no versions of syncfusion_flutter_charts match >20.4.41 <21.0.0, syncfusion_flutter_charts ^20.4.41 requires intl ^0.17.0.
> So, because track_my_indoor_exercise depends on both intl ^0.18.0 and syncfusion_flutter_charts ^20.4.41, version solving failed.
> pub get failed (1; So, because track_my_indoor_exercise depends on both intl ^0.18.0 and syncfusion_flutter_charts ^20.4.41, version solving failed.)

To be honest I don't quite understand the error, because `^0.17.0` supposed to allow any version which is at least 0.17.0 (including 0.18.0) but less than 1.0.0. But the `pub get` fails.

**Comment by MrCsabaToth** [comment]
Extra info: I'm on the stable Flutter channel.

```
flutter doctor -v
[✓] Flutter (Channel stable, 3.3.10, on Devuan GNU/Linux 5 (daedalus/ceres) 6.0.0-6-amd64, locale en_US.UTF-8)
    • Flutter version 3.3.10 on channel stable at /home/csaba/flutter
    • Upstream repository https://github.com/flutter/flutter
    • Framework revision 135454af32 (2 weeks ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /home/csaba/Android/Sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /home/csaba/android-studio-preview/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Linux toolchain - develop for Linux desktop
    • Debian clang version 14.0.6
    • cmake version 3.25.1
    • ninja version 1.11.1
    • pkg-config version 1.8.0

[✓] Android Studio (version 2021.3)
    • Android Studio at /home/csaba/android-studio-preview
    • Flutter plugin version 71.2.3
    • Dart plugin version 213.7433
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] Android Studio
    • Android Studio at /home/csaba/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.74.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.56.0

[✓] Connected device (1 available)
    • Linux (desktop) • linux • linux-x64 • Devuan GNU/Linux 5 (daedalus/ceres) 6.0.0-6-amd64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by MrCsabaToth** [comment]
Wow, `pub upgrade` is even weirder:
> Because syncfusion_flutter_charts >=19.1.54 depends on intl ^0.17.0 and track_my_indoor_exercise depends on intl ^0.18.0, syncfusion_flutter_charts >=19.1.54 is forbidden.
> So, because track_my_indoor_exercise depends on syncfusion_flutter_charts ^20.4.41, version solving failed.
> pub finished with exit code 1

I have no idea where it pulls this `19.1.54` version number from, that's super old.

**Comment by MrCsabaToth** [comment]
Anyone else has this problem? I'm still dealing with it, holds back versions.

**Comment by Yuvaraj-Gajaraj** [comment]
Hi @MrCsabaToth 

The reported issue regarding the collision occurring while having both syncfusion_flutter_charts and third-party widgets with intl package version 0.18.0, has been fixed and rolled out in our weekly patch release. To avoid this please upgrade the chart package to the latest version below.



Version: https://pub.dev/packages/syncfusion_flutter_charts/versions/20.4.44

**Comment by MrCsabaToth** [comment]
It's fixed with the 20.4.44
